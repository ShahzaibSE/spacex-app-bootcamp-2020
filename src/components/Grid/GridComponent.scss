.grid-container {
    // margin-left:10px;
    // margin-right:10px;
    // margin:0 auto;
}