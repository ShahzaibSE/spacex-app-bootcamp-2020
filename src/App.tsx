import React from 'react';
import './App.css';
// Components.
import GridComponent from "./components/Grid/Grid"

function App() {
  return (
    <div className="App">
      <GridComponent />
    </div>
  );
}

export default App;
