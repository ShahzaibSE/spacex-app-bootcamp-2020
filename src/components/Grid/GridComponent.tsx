import React from 'react'
import Grid from "@material-ui/core/Grid"
// Components.
import LaunchListContainerComponent from "../LaunchList/LaunchList.container"
import Header from "../Header/Header"
// Assets.
import "./GridComponent.scss"

const GridComponent = () => {
    return (
        <div>
            <Grid container>
                <Grid item sm={12} md={12} lg={12}>
                    <Header/>
                </Grid>
            </Grid>
        </div>
    )
}

export default GridComponent
