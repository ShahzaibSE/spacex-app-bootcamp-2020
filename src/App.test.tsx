import React from "react"
import {shallow, mount} from "enzyme"
// Component.
import GridComponent from "./components/Grid/GridComponent"
import App from "./App"

describe("App root component", ()=>{
    let container: any;

    beforeEach(()=>(
        container = shallow(<App/>)
    ))

    it("testing UI of <App/>", ()=>{
        expect(container).toMatchSnapshot()
    })

    it("at least one div to be found", ()=>{
        expect(container.find("div").length).toEqual(1)
    })

    it("check if Grid component has been rendered", ()=>{
        expect(container.containsMatchingElement(<GridComponent/>)).toEqual(true)
    })
})