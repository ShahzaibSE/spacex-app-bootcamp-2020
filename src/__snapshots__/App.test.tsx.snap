// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App root component check if Grid component renders 1`] = `true`;
